# windshape configuration file

# Uses Graphical User Interface
gui: True

# Number of log files
max_files: 10

# Fans array
fansarray:
  db_update_rate: 10.0
  broadcast_port: 58084
  auto_wind: False

# Drone tracking using mocap system
tracking:
  timeout: 0.1
  tracker: "drone"
  refresh: 1.0

# Drone real time management
update:
  control_rate: 60.0
  mocap_sending_rate: 120.0
  record_rate: 20.0
  service_wait_timeout: 2.0
  update_rate: 10.0

# Drone default control mode
control:

  # Wind
  pwm_gain: 100
  min_wind: 5
  max_wind: 100
  
  # Setpoint
  setpoint: [0, 0, 0.5, 0]
  
  # Target
  target: "target"
  follow: True
  mimic: False
  shift: [0, -1.0, 0, 0, 0, 0]
  
  # Control mode
  offboard: True
  mask: [False, False, False, False]

  # Drone offboard PID attitude control
  sp_filter: 200
  d_filter: 0.5

  pid:
    roll:
      kp: 0.8
      ki: 0.1
      kd: 0.35
      min: deg(-30)
      max: deg(30)
      ff: deg(0)
    pitch:
      kp: 0.8
      ki: 0.1
      kd: 0.35
      min: deg(-30)
      max: deg(30)
      ff: deg(0)
    yaw:
      kp: 1.0
      ki: 0.1
      kd: 0.05
      min: deg(-180)
      max: deg(180)
      ff: deg(0)
    thrust:
      kp: 1.0
      ki: 0.35
      kd: 0.15
      min: 0.0
      max: 0.8
      ff: 0.35
